gst/nnstreamer/tensor_filter/tensor_filter_common.c

```diff
@@ -680,6 +680,120 @@ gst_tensor_filter_common_free_property (GstTensorFilterPrivate * priv)
 }
 
+/**
+ * @brief Get available framework from given file when user selects auto option
+ * @param[in] model_files the prediction model paths
+ * @param[in] num_models the number of model files
+ * @return Automatically detected framework name
+ */
+gchar *
+gst_tensor_filter_framework_auto_detection (const gchar ** model_files,
+    unsigned int num_models)
+{
+  const GstTensorFilterFramework *fw = NULL;
+  gchar *file_extension, *file_extension_2;
+  gchar *fw_priority_str = NULL, *detected_fw = NULL;
+  gchar **fw_priority_str_arr;
+  guint i;
+
+  file_extension = strrchr (model_files[0], '.');
+
+  /* Detect framework based on file extension */
+  if (g_strcmp0 (file_extension, ".pb") == 0)
+    fw_priority_str =
+        nnsconf_get_custom_value_string ("filter", "framework_priority_pb");
+  else if (g_strcmp0 (file_extension, ".tflite") == 0)
+    fw_priority_str =
+        nnsconf_get_custom_value_string ("filter", "framework_priority_tflite");
+  else if (g_strcmp0 (file_extension, ".graph") == 0)
+    detected_fw = g_strdup ("movidius-ncsdk2");
+  else if (g_strcmp0 (file_extension, NNSTREAMER_SO_FILE_EXTENSION) == 0)
+    detected_fw = g_strdup ("custom");
+  else if (g_strcmp0 (file_extension, ".bin") == 0
+      || g_strcmp0 (file_extension, ".xml") == 0)
+    detected_fw = g_strdup ("openvino");
+  else if (num_models == 2) {
+    file_extension_2 = strrchr (model_files[1], '.');
+    if ((g_strcmp0 (file_extension, ".prototxt") == 0
+            && g_strcmp0 (file_extension_2, ".caffemodel") == 0)
+        || (g_strcmp0 (file_extension_2, ".prototxt") == 0
+            && g_strcmp0 (file_extension, ".caffemodel") == 0))
+      detected_fw = g_strdup ("snap");
+    else if ((g_strcmp0 (file_extension, ".so") == 0
+            && g_strcmp0 (file_extension_2, ".nb") == 0)
+        || (g_strcmp0 (file_extension_2, ".so") == 0
+            && g_strcmp0 (file_extension, ".nb") == 0))
+      detected_fw = g_strdup ("vivante");
+  }
+
+  /* .pb or .tflite have multiple framework. */
+  if (fw_priority_str != NULL) {
+    fw_priority_str_arr = g_strsplit (fw_priority_str, ",", -1);
+
+    for (i = 0; i < g_strv_length (fw_priority_str_arr); i++) {
+      fw = nnstreamer_filter_find (fw_priority_str_arr[i]);
+
+      if (fw) {
+        detected_fw = g_strdup (fw_priority_str_arr[i]);
+        break;
+      } else {
+        g_debug ("Cannot identify the neural network framework, %s\n",
+            fw_priority_str_arr[i]);
+      }
+    }
+    if (fw == NULL)
+      g_warning ("Cannot get any neural network framework of %s, \n",
+          file_extension);
+
+    g_free (fw_priority_str);
+    g_strfreev (fw_priority_str_arr);
+  }
+
+  g_debug ("Automatically detected framework is %s.\n", detected_fw);
+  g_debug
+      ("If you want to change priority of framerwork for auto detection, \n");
+  g_debug ("Please edit meson_option.txt \n");
+  g_debug ("You can find the file at root path of nnstreamer. \n");
+
+  return detected_fw;
```

Comment:
At 792, it seems that you just print out a warning message using g_warn. Here is the question. Is it OK with returning NULL? Are there any fallback mechanisms?

Reply:
Yes, it just print out warning message.
If NULL is returned, GStreamer will generate an error later.